src/main/java/com/google/devtools/build/lib/skyframe/BaselineOptionsFunction.java

```diff
@@ -0,0 +1,92 @@
+// Copyright 2023 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.devtools.build.lib.skyframe;
+
+import com.google.devtools.build.lib.analysis.PlatformOptions;
+import com.google.devtools.build.lib.analysis.config.BuildOptions;
+import com.google.devtools.build.lib.analysis.config.CoreOptions;
+import com.google.devtools.build.lib.analysis.config.ExecutionTransitionFactory;
+import com.google.devtools.build.lib.analysis.config.transitions.BaselineOptionsValue;
+import com.google.devtools.build.lib.analysis.config.transitions.PatchTransition;
+import com.google.devtools.build.lib.analysis.config.transitions.TransitionUtil;
+import com.google.devtools.build.lib.cmdline.Label;
+import com.google.devtools.build.skyframe.SkyFunction;
+import com.google.devtools.build.skyframe.SkyFunctionException;
+import com.google.devtools.build.skyframe.SkyKey;
+import com.google.devtools.build.skyframe.SkyValue;
+import com.google.devtools.common.options.OptionsParsingException;
+import javax.annotation.Nullable;
+
+/** A builder for {@link BaselineOptionsValue} instances. */
+public final class BaselineOptionsFunction implements SkyFunction {
+  @Override
+  @Nullable
+  public SkyValue compute(SkyKey skyKey, Environment env)
+      throws InterruptedException, BaselineOptionsFunctionException {
+    BaselineOptionsValue.Key key = (BaselineOptionsValue.Key) skyKey.argument();
+
+    BuildOptions rawBaselineOptions = PrecomputedValue.BASELINE_CONFIGURATION.get(env);
+
+    // Some test infrastructure only creates mock or partial top-level BuildOptions such that
+    // PlatformOptions or even CoreOptions might not be included.
+    // In that case, is not worth doing any special processing of the baseline.
+    if (!rawBaselineOptions.getFragmentClasses().contains(PlatformOptions.class)
```

Comment:
On master this is `rawBaselineOptions.hasNoConfig()`, which was introduced by https://cs.opensource.google/bazel/bazel/+/ea049e46c84fdcef2e1acd83d07053b7fa0a5cb5. I "inlined" it based on the comment, but please let me know whether you would want me to cherry-pick the full function into `BuildOptions`.